### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by deock90** [CGCookie/retopoflow#1101]
im using a wacom with retopoflow, and it crashed.

version: blender 3.0
version retopoflow: 3.26

I have to close blender to work on my project again, Retopoflow's autosave works fine, but you have to close blender and reopen it, because it won't even let me exit Retopoflow.

![error](https://user-images.githubusercontent.com/41754489/158277247-ae180dae-6711-479c-b1a4-7ca6a24ce130.png)

**Comment by vxlcoder** [comment]
thanks for the report, @deock90 .

I have a good idea what the problem is and will get it fixed asap.

**Comment by TekuConcept** [comment]
I think sporadic MMB events are tripping up the addon. With the above patch in place, I can trigger another exception in retopoflow that freezes blender using sporadic [MMB, CTRL, Move] input events.

```
____________________________________________________________________________________________________
Exception Info
- Time: 2022-04-02 19:41:53.219253
- Count: 1
- Hash: aa91525e6e6bc87f4600a32e1190d12e
- Info:
  - EXCEPTION (<class 'TypeError'>): 'NoneType' object is not subscriptable
  -     \scripts\modules\bpy_extras\view3d_utils.py
  - 000 0171:location_3d_to_region_2d() prj = rv3d.perspective_matrix @ Vector((coord[0], coord[1], coord[2], 1.0))
  -     \scripts\addons\RetopoFlow\retopoflow\rf\rf_spaces.py
  - 001 0091:Point_to_Point2D() xy = location_3d_to_region_2d(self.actions.region, self.actions.r3d, xyz)
  -     \scripts\addons\RetopoFlow\retopoflow\rftool_polypen\polypen.py
  - 002 0721:<listcomp>() coords = [self.rfcontext.Point_to_Point2D(co) for co in coords]
  - 003 0721:draw_lines() coords = [self.rfcontext.Point_to_Point2D(co) for co in coords]
  - 004 0872:draw_postpixel() self.draw_lines([co1, co2, co3, co0])
  -     .../common\fsm.py
  - 005 0135:wrapped() return fn(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by TekuConcept** [comment]
In file, `retopoflow/rf/rf_spaces.py`, adding `if xyz is None: return None` to `Point_to_Point2D` seems to fix the follow-up bug.

```py
    def Point_to_Point2D(self, xyz:Point):
        if xyz is None: return None # <-- PATCH HERE
        xy = location_3d_to_region_2d(self.actions.region, self.actions.r3d, xyz)
        if xy is None: return None
        return Point2D(xy)
```

**Comment by vxlcoder** [comment]
sorry for the delayed reply, @TekuConcept, and great job with the fix :)  I've added your patch in commit 2390764.

closing this issue.  let us know if you have any other issues.

happy retopo'ing!
